internal/impl/update.go

```diff
@@ -0,0 +1,46 @@
+package impl
+
+import (
+	"context"
+	"fmt"
+)
+
+func (d *Devbox) Update(ctx context.Context, pkgs ...string) error {
+	pkgsToUpdate := []string{}
+	if len(pkgs) == 0 {
+		pkgsToUpdate = append([]string(nil), d.packages()...)
+	} else {
+		for _, pkg := range pkgs {
+			found, err := d.findPackageByName(pkg)
+			if err != nil {
+				return err
+			}
+			pkgsToUpdate = append(pkgsToUpdate, found)
+		}
+	}
+
+	for _, pkg := range pkgsToUpdate {
+		if !d.lockfile.IsVersionedPackage(pkg) {
+			fmt.Fprintf(d.writer, "Skipping %s because it is not a versioned package\n", pkg)
```

Comment:
Not sure. Do we want the full advice to be something like:

"Skipping %s because it is not a versioned package, and so we derive its version from the nixpkgs commit hash in devbox.json. Consider making it a versioned package, or changing the nixpkgs commit hash to get a newer version.\n"

Reply:
If we do want to give this longer advice, I'd suggest buffering the skipped packageNames, and printing this advice once for all of them at the end of the entire `update` execution.

Reply:
The idea was to print one line per package to give you an idea of status. Don't disagree that we should explain to user how to transition but not sure this is the best place.

Generally I lean to less explaining and just more telling/fixing for example:

```
Warning: your devbox.json contains legacy unversioned packages. 
Please run `devbox config upgrade` to fix.
```

Reply:
Also:

* `nixpkgs commit hash` is not really something non-nixers will be familiar with.
* I'm not sure I want to encourage people to mess with the hash and instead they should migrate to versioned packages.